[LiveMetricsExporter] Add Filtering Support part 7 - Sync Model change for Duration

xiang17 opened the pull request:
PR following up on #41523.

Models for all telemetry's `Duration` field changed from `TimeSpan` to `string`. Changing the code to parse the field to `TimeSpan` to facilitate comparisons actions later.

To be specific, these two `DocumentIngress` implementations have `Duration` field but is of type `string`:
https://github.com/Azure/azure-sdk-for-net/blob/9ce35d5370ddef0707776075d855c6b290bbc0d3/sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Generated/Models/Request.cs#L45
https://github.com/Azure/azure-sdk-for-net/blob/9ce35d5370ddef0707776075d855c6b290bbc0d3/sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Generated/Models/RemoteDependency.cs#L45

xiang17 committed:
Parse the Duration field from string to TimeSpan to update with Model change

rajkumar-rangaraj commented:
Is there an actual issue, or are you anticipating one? If there is an actual issue, can you explain that in detail in the PR description?

xiang17 committed:
Only convert the fieldExpression if the incoming data field type of the fieldExpression is of type `string`. In other words, the conversion should only happen if Document.Duration field type is `string`. If there are other field type of `TimeSpan`, don't do this conversion.

xiang17 committed:
Add unit tests that checks when Duration field is of type `string`

xiang17 commented:
> Is there an actual issue, or are you anticipating one? If there is an actual issue, can you explain that in detail in the PR description?

I've updated the PR description and added some unit tests.
